core/src/main/java/org/apache/calcite/rel/rel2sql/RelToSqlConverter.java

```diff
@@ -332,14 +332,14 @@ public Result visit(Filter e) {
       final Builder builder = x.builder(e);
       SqlNode condition = builder.context.toSql(null, e.getCondition());
       SqlNode existHaving = x.asSelect().getHaving();
-      if (existHaving != null) {
+      if (existHaving == null) {
+        builder.setHaving(condition);
+      } else {
```

Comment:
you could replace this whole `if` block with a call to `SqlUtil.andExpressions`

Reply:
done.